opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -114,6 +115,77 @@ def shutdown(self) -> None:
             sp.shutdown()
 
 
+class EventBase(abc.ABC):
+    def __init__(self, name: str, timestamp: Optional[int] = None) -> None:
+        self._name = name
+        if timestamp is None:
+            self._timestamp = time_ns()
+        else:
+            self._timestamp = timestamp
+
+    @property
```

Comment:
Using properties in the base class means implementations can't use regular attributes. Not a big deal if we don't expect users to extend `EventBase` and  `LinkBase` though.